barretenberg/cpp/src/barretenberg/vm2/simulation/events/alu_event.hpp

```diff
@@ -14,12 +14,13 @@ enum class AluOperation {
 
 struct AluEvent {
     AluOperation operation;
-    MemoryAddress a_addr;
-    MemoryAddress b_addr;
-    MemoryAddress dst_addr;
     MemoryValue a;
     MemoryValue b;
-    MemoryValue res;
+    MemoryValue c;
+    // Only need single tag info here (check this for MOV or CAST )
+    // For operations that have a specific output tag (e.g., EQ/LT), the output tag is unambiguous
+    // We still might prefer to include tags per operands to simply tracegen...
+    MemoryTag tag;
 };
```

Comment:
I've also been thinking about the selectors and lookups.

In execution, we probably only want selectors to know which gadget to use, so sel_alu, etc, and NOT sel_add. Then we only pass the (execution) _opcode_ to the alu gadget, and the ALU can, from that, get its sel_add, etc. I think this minimizes duplication of columns.

So, we'll also need an ExecutionOpcode here or, if we want to be type-safe, some AluOperation enum. However simulation should not pass this. Each method knows its opcode (add() knows it has to emit ADD).

Reply:
(do we want to to add part of the above as a comment or at least get it to Miranda?)

Reply:
pre-emptively tagging @MirandaWood here in case this comes up 😁